woocommerce/wc-admin
# Add required dependencies for pnpm nx build {package} command

Fixes #8433 

This PR adds required dependencies for the packages in `/packages` to make them work with `pnpm nx build {package}` in The Merge environment. 

- Also changed some imports in `@woocommerce/components` to the relative path to run the `test` command independently.

### Detailed test instructions:

1. Refer to `archives/C01SFMVEYAK/p1646777260567079?thread_ts=1646772045.138809&cid=C01SFMVEYAK` slack thread to setup The Merge environment. (You might need to run `brew install git-filter-repo` if you don't have `git-filter-repo` in your local.)
2. Once you have The Merge environment,  run `pnpm run clean && rm -r packages/**/node_modules` in WCA repo and copy `packages/*` from this branch to `packages/js/` in The Merge. 
3. Run `pnpm install` in The Merge.
4. Run the following commands in The Merge and confirm the commands work without an error.

```bash
pnpm nx run-many --target=build --all --exclude=@woocommerce/api,@woocommerce/api-core-tests,@woocommerce/e2e-core-tests,@woocommerce/e2e-environment,@woocommerce/e2e-utils,woocommerce,woocommerce-legacy-assets,@woocommerce/dependency-extraction-webpack-plugin
```

Copy the `config/development.json` from WCA to `packages/config/development.json` in the Merge.

```bash
pnpm nx run-many --target=test --all --exclude=@woocommerce/api,@woocommerce/api-core-tests,@woocommerce/e2e-core-tests,@woocommerce/e2e-environment,@woocommerce/e2e-utils,woocommerce,woocommerce-legacy-assets,@woocommerce/dependency-extraction-webpack-plugin,@woocommerce/admin-e2e-tests,@woocommerce/onboarding,@woocommerce/js-tests, @woocommerce/notices,@woocommerce/tracks
```

no changelog
